src/replication.c

```diff
@@ -1314,7 +1371,8 @@ check_for_command_completion(spec_t *spec, rcommon_cmd_t *rcomm_cmd, ISTGT_LU_CM
 {
 	int i, rc = 0;
 	uint8_t *data = NULL;
-	int success = 0, failure = 0;
```

Comment:
Few edge cases I can think of are (for both read and write):
- CF, RF as 1 i.e., single replica case where it might be healthy or degraded
- status of replicas changes or number of replicas connected changes after IOs are sent
- similar to above thing, status of replicas changes or number of replicas connected changes after verifying for spec's ready flag. One thing to consider here is: all replicas are disconnected after verifying spec's ready flag, so, there are no replicas at all. So, copies_sent would be 0.

Reply:
1. CF, RF as 1 i.e., single replica case where it might be healthy or degraded
> *In this case, `min_response` and `copies_sent` will be 1. In read case, if `success == 1`*
*then we will reply with `rc=1` or if `response_received == rcomm_cmd->copies_sent` then*
*we will reply with `rc = -1`.*
2. status of replicas changes or number of replicas connected changes after IOs are sent
> *We are taking spec lock while assigning IOs to replica and during error case handling for*
*the replica. So, it will not create the problem in this case.*
3. status of replicas changes or number of replicas connected changes after verifying for spec's ready flag
> *same as in 2nd case*